### PR TITLE
Not all valid IPv6 addresses can be generated

### PR DESCRIPTION
**Pull request by zinki97**
Addresses like a::a can't be generated

### CODE REVIEW

**Comment by zinki97** [comment]
Ready for review

File: engine/src/test/java/net/jqwik/api/ArbitrariesEmailsTests.java [hunk 1]
```diff
@@ -116,7 +115,7 @@ void tldNotAllNumeric(@ForAll("emails") String email) {
 			assertThat(stringContainsMinimumOneChar(ALLOWED_NOT_NUMERIC_CHARS_TLD, tld)).isTrue();
 		}
 
-		@Property
+		@Property(tries = 5000)
```
Comment (jlink) [inline engine/src/test/java/net/jqwik/api/ArbitrariesEmailsTests.java:118]: Why would you try that 5000 times?

**Review by jlink — APPROVED** [review]

File: engine/src/test/java/net/jqwik/api/ArbitrariesEmailsTests.java (hunk 1 above)
Comment (zinki97) [reply engine/src/test/java/net/jqwik/api/ArbitrariesEmailsTests.java:118]: To have a greater chance that addresses like a::a are generated

File: engine/src/test/java/net/jqwik/api/ArbitrariesEmailsTests.java [hunk 2]
```diff
@@ -313,11 +316,20 @@ void ipv6Shrinking(@ForAll Random random) {
 			String value = shrinkToMinimal(emails, random, falsifier);
 			String domain = getDomainOfEmail(value);
 			domain = domain.substring(1, domain.length() - 1);
-			String[] domainParts = domain.split(":");
-			assertThat(isValidIPv6Address(domain)).isTrue();
-			IntStream.range(0, domainParts.length).forEach(i -> {
-				assertThat(domainParts[i]).isIn("0", "");
-			});
+			assertThat(domain).isEqualTo("::");
```
Comment (jlink) [inline engine/src/test/java/net/jqwik/api/ArbitrariesEmailsTests.java:319]: parts of ipv6 should always shrink to "0", should they not?

**Pull request merged**